addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -4269,10 +4269,14 @@ export class OdooEditor extends EventTarget {
             if (fragment.hasChildNodes()) {
                 this._applyCommand('insert', fragment);
             }
-        } else if (files.length && targetSupportsHtmlContent) {
-            this.addImagesFiles(files).then(html => this._applyCommand('insert', this._prepareClipboardData(html)));
-        } else if (clipboardHtml && targetSupportsHtmlContent) {
-            this._applyCommand('insert', this._prepareClipboardData(clipboardHtml));
+        } else if ((files.length || clipboardHtml) && targetSupportsHtmlContent) {
+            // Differentiate or choose between images and html
```

Comment:
I think this comment should refer to the fact that we treat table as a special case. We know it because it's the point of the task and it's obvious from the code itself, but I think it would look so weird to any future reader that it warrants its own comment in the code. The comment should explicitely mention the fact that we treat html tables differently and explain why we need to do that. (Please don't forget the dot at the end of the new comment :))